content/intro-to-storybook/ember/en/screen.md

```diff
@@ -63,78 +76,106 @@ export const reducer = (state, action) => {
-        <div class="title-message">
-          Oh no!
-        </div>
-        <div class="subtitle-message">
-          Something went wrong
+{{!--app/components/inbox-screen.hbs--}}
+<div>
+  <div class="page lists-show">
+    <nav>
+      <h1 class="title-page">
+        <span class="title-wrapper">
+          Taskbox
+        </span>
+      </h1>
+    </nav>
+    {{#if this.loading}}
+      <LoadingRow />
+      <LoadingRow />
+      <LoadingRow />
+      <LoadingRow />
+      <LoadingRow />
+    {{else if this.error}}
+      <div class="page lists-show">
+        <div class="wrapper-message">
+          <span class="icon-face-sad"></span>
+          <div class="title-message">
+            Oh no!
+          </div>
+          <div class="subtitle-message">
+            Something went wrong
+          </div>
         </div>
       </div>
-    </div>
-  {{else}}
-    <TaskList />
-  {{/if}}
+    {{else}}
+      <TaskList
+        @tasks={{this.tasks}}
+        @pinTask={{this.pinTask}}
+        @archiveTask={{this.archiveTask}}
+      />
+    {{/if}}
+  </div>
 </div>
 ```
 
-Then, we can create a container, which again grabs the data for the `PureInboxScreen` in `app/components/inbox-screen.hbs` add the following:
-
-```handlebars
-{{!--app/components/inbox-screen.hbs --}}
-<div>
-  <PureInboxScreen @error={{this.error}} />
-</div>
-```
+It already has sections to show various states for loading and errors. Next up
+is to move our already existing data loading from the routes and controller in
+the previous step into the component:
 
 In `app/components/inbox-screen.js` add the following:
 
 ```javascript
 // app/components/inbox-screen.js
 import Component from '@glimmer/component';
-import { connect } from 'ember-redux';
+import { action } from '@ember/object';
+import { store } from '../store';
+import { pinTask, archiveTask } from '../actions';
 
-const stateToComputed = state => {
-  const { reducer } = state;
-  const { isError } = reducer;
+export default class InboxScreenComponent extends Component {
+  get loading() {
+    return this.args.loading ?? store.getState().isLoading;
+  }
 
-  return {
-    error: isError,
-  };
-};
+  get error() {
+    return this.args.error ?? store.getState().isError;
+  }
+
+  get tasks() {
+    return store.getState().tasks.filter(
+      (t) => t.state === 'TASK_INBOX' || t.state === 'TASK_PINNED'
+    );
+  }
+
+  @action
+  pinTask(task) {
+    store.dispatch(pinTask(task.id));
+  }
 
-class InboxScreenComponent extends Component {}
-export default connect(stateToComputed)(InboxScreenComponent);
+  @action
+  archiveTask(task) {
+    store.dispatch(archiveTask(task.id));
+  }
+}
 ```
 
-We also change the `application` template to render the `InboxScreen` (eventually we would use a router to choose the correct screen, but let's not worry about that here):
+We will remove the `tasks` route from the previous step and mount the
+`InboxScreen` in the `application` template.
```

Comment:
@gossi we could probaby phrase this a bit different. We could probably go with:

"At this stage we can now safely delete the route and controller we've setup earlier and add our `InboxScreen` directly in the `application` template." What do you think?